### PR TITLE
ToolsPanel: Update typography panel layout

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
Depends on:
- https://github.com/WordPress/gutenberg/pull/35621

Related:
- https://github.com/WordPress/gutenberg/pull/33744

## Description

The `ToolsPanel` originally relied upon providing a hardcoded `single-column` class name to allow tools panel items to have an easy and consistent means of applying styles to span a single column. 

The `single-column` class will be removed as part of https://github.com/WordPress/gutenberg/pull/35621 which updates the `ToolsPanel` to use the `Grid` component to handle its layout internally. 

This PR updates the Typography panel styles to handle this. It also tweaks the block editor's letter spacing control to allow opting out of the `max-width` so that it fills a column in the `ToolsPanel` as per other controls.

## How has this been tested?

1. Checkout this PR, create a post, add a site title block and select it (site title opts into most typography features)
2. Toggle on all the typography controls from the typography block supports panel in the sidebar
3. Ensure the panel's controls are laid out correctly and the letter-spacing control is full width

## Screenshots 

| Before | After |
|---|---|
| Screen Shot 2021-10-25 at 3 29 10 pm | Screen Shot 2021-10-25 at 3 28 13 pm |

## Types of changes
Enhancement. Bug fix.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by stacimc — APPROVED** [review]
Tests well for me!

All fields look correct when added in the block editor, including full-width letter spacing:
Screen Shot 2021-10-25 at 2 17 01 PM

And everything looks good in Global styles as well, with letter spacing matching the appearance of the line height field:
Screen Shot 2021-10-25 at 2 15 54 PM

File: packages/block-editor/src/components/letter-spacing-control/index.js [hunk 1]
```diff
@@ -15,12 +15,17 @@ import useSetting from '../../components/use-setting';
...
+ * @param {string}   props.value        Currently selected letter-spacing.
+ * @param {Function} props.onChange     Handles change in letter-spacing selection.
+ * @param {boolean}  props.withMaxWidth Whether to restrict the field's max width.
+ * @return {WPElement}                  Letter-spacing control.
  */
-export default function LetterSpacingControl( { value, onChange } ) {
+export default function LetterSpacingControl( {
+	value,
+	onChange,
+	withMaxWidth = true,
```
Comment (mirka) [inline packages/block-editor/src/components/letter-spacing-control/index.js]: I imagine this kind of thing will be a common need and I'm not sure that exposing a new prop like `withMaxWidth` is a pattern we want to adopt widely. Would it be safer (i.e. more conservative) for now to simply expose `__unstableInputWidth` and pass that down to `UnitControl`? Then a consumer can override it with a falsy value.

```suggestion
	__unstableInputWidth = '60px',
```

Comment (aaronrobertshaw) [reply packages/block-editor/src/components/letter-spacing-control/index.js]: Thanks @mirka, I wasn't sure about passing through the unstable prop. I do like that approach better so have pushed the suggested change.

**Review by glendaviesnz — APPROVED** [review]
This works as advertised for me - got same results as Staci in block settings and global styles

File: packages/block-editor/src/components/letter-spacing-control/index.js [hunk 2]
```diff
@@ -15,12 +15,18 @@ import useSetting from '../../components/use-setting';
...
+ * @param {Function} props.onChange             Handles change in letter-spacing selection.
+ * @param {boolean}  props.__unstableInputWidth Input width to pass through to inner UnitControl.
+ *
+ * @return {WPElement}                          Letter-spacing control.
  */
-export default function LetterSpacingControl( { value, onChange } ) {
+export default function LetterSpacingControl( {
+	value,
+	onChange,
+	__unstableInputWidth = '60px',
```
Comment (ciampo) [inline packages/block-editor/src/components/letter-spacing-control/index.js:28]: The JSDocs mark `__unstableInputWidth` as of type `boolean`, but the default value is `60px` — is this on purpose?

File: packages/block-editor/src/components/letter-spacing-control/index.js [hunk 3]
```diff
@@ -15,12 +15,18 @@ import useSetting from '../../components/use-setting';
...
  * Control for letter-spacing.
  *
- * @param {Object}   props          Component props.
- * @param {string}   props.value    Currently selected letter-spacing.
- * @param {Function} props.onChange Handles change in letter-spacing selection.
- * @return {WPElement}                      Letter-spacing control.
+ * @param {Object}   props                      Component props.
+ * @param {string}   props.value                Currently selected letter-spacing.
+ * @param {Function} props.onChange             Handles change in letter-spacing selection.
+ * @param {boolean}  props.__unstableInputWidth Input width to pass through to inner UnitControl.
```
Comment (ciampo) [inline packages/block-editor/src/components/letter-spacing-control/index.js:21]: By looking at https://github.com/WordPress/gutenberg/blob/0160b7f1ccdc806a64eb111b36b4fd9c328ad49a/packages/components/src/input-control/types.ts, it seems like this prop should be of type `string | number | undefined` (and not `boolean`) — is this on purpose?

File: packages/block-editor/src/hooks/letter-spacing.js [hunk 4]
```diff
@@ -45,6 +45,7 @@ export function LetterSpacingEdit( props ) {
 		<LetterSpacingControl
 			value={ style?.typography?.letterSpacing }
 			onChange={ onChange }
+			__unstableInputWidth={ false }
```
Comment (ciampo) [inline packages/block-editor/src/hooks/letter-spacing.js:48]: Similarly, I'm not sure if passing `false` is correct here?

**Review by ciampo — COMMENTED** [review]
A bit late to the party, but I left a couple of comments that may require a follow-up PR :)

File: packages/block-editor/src/components/letter-spacing-control/index.js (hunk 3 above)
Comment (aaronrobertshaw) [reply packages/block-editor/src/components/letter-spacing-control/index.js:21]: I've created follow-up PR to address this: https://github.com/WordPress/gutenberg/pull/36367.

To correct the `boolean` type here but still override the default I needed to use `null`. For backwards compatibility, we need to provide a default width so setting the `__unstableWidth` prop to `undefined` doesn't work when wishing to restore full width to the control under the Typography `ToolsPanel`.

File: packages/block-editor/src/hooks/letter-spacing.js (hunk 4 above)
Comment (aaronrobertshaw) [reply packages/block-editor/src/hooks/letter-spacing.js:48]: I've [switched this to `null`](https://github.com/WordPress/gutenberg/pull/36367/files#diff-7f6873e89e742b9f974a2b72f19ac8dc16b704d513aa0c2d31e29be331519c95R48) instead to override the default width required for backwards compatibility purposes.

**Pull request merged**